README.md

```diff
@@ -27,91 +27,177 @@ from the Sentry SDKs as well as the Sentry server into a proxy process.
+make build
 
-Stats can be submitted to a statsd server by configuring `metrics.statsd` key.
-It can be put to a `ip:port` tuple. Additionally `metrics.prefix` can be
-configured to have a different prefix (the default is `sentry.relay`). This
-prefix is added in front of all metrics.
+# Build with release optimizations and debug information.
+make release
+```
 
-## License
+For quickly verifying that Relay compiles after making some changes, you can
+also use `cargo check`:
 
-Like Sentry, Relay is licensed under the BSL. See the `LICENSE` file and [this
-forum post](https://forum.sentry.io/t/re-licensing-sentry-faq-discussion/8044)
-for more information.
+```bash
+cargo check --all --all-features
+```
 
-## Development
+### Features
 
-We're going to settle on using vscode for this project for now. We're targeting
-stable rust at the moment and the repo is appropriately configured.
+By default, Relay compiles without _processing_ mode. This is the configuration
+used for Relays operating as proxys. There are two optional features:
 
-If you have `systemfd` and `cargo-watch` installed, the `make devserver` command can auto-reload the
-relay:
+- **`processing`**: Enables event processing and ingestion functionality. This
+  allows to enable processing in config. When enabled, Relay will produce events
+  into a Kafka topic instead of forwarding to the configured upstream. Also, it
+  will perform full event normalization, filtering, and rate limiting.
 
-    $ cargo install systemfd cargo-watch
-    $ make devserver
+- **`with_ssl`**: Enables SSL support in the Server.
 
+To enable a feature, pass it to the cargo invocation. For example, to run tests
+across all workspace crates with the `processing` feature enabled, run:
 
-### SSL
+```bash
+cargo run --features=processing
+```
 
-The repository contains a SSL-certificate + private key for development
-purposes. It comes in two formats: Once as a `(.pem, .cert)`-pair, once as
-`.pfx` (PKCS #12) file.
+### Tests
 
-The password for the `.pfx` file is `password`.
+The test suite comprises unit tests, an integration test suite and a separate
+test suite for the Python package. Unit tests are implemented as part of the
+Rust crates and can be run via:
+
+```bash
+# Tests for default features
+make test-rust
 
-### Running in Docker
+# Run Rust tests for all features
+make test-rust-all
+```
 
-Docker image for `relay` can be found at `us.gcr.io/sentryio/relay`.
+The integration test suite requires `python`. By default, the integration test
+suite will create a virtualenv, build the Relay binary with processing enabled,
+and run a set of integration tests:
 
-For example, you can start the latest version of `relay` as follows:
+```bash
+# Build Relay and run integration tests
+make test-integration
 
-```sh
-docker run -v $(pwd)/configs/:/etc/relay/ us.gcr.io/sentryio/relay run --config /etc/relay
+# Build and run a single test manually
+make build
+pytest tests/integration -k <test_name>
```

Comment:
This requires a venv that has all deps installed and that the venv is activated. Assuming the user ran tests before, I would replace this with:

```suggestion
make build
.venv/bin/pytest tests/integration -k <test_name>
```

I think we might want to have a make target that sets up the virtualenv and installs dependencies, currently there is no such target that does just that and nothing more.

Reply:
That's a good idea. I wish we only had to set up the virtual env once, but the problem is that we need different requirements for the different targets. We also run Python2 tests which don't support most of the tools, like `black` and so on.

We could add a `make setup-integration-requirements`, but I'm not sure how useful that is given that people will almost always run `make test-integration` first.